### PR TITLE
Complete App till Dashboard

### PR DESCRIPTION
**Pull request by akashsarkar188**

### CODE REVIEW

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/activities/Splash.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_splash);
+
+        try {
+            getWindow().setNavigationBarColor(ContextCompat.getColor(Splash.this, R.color.colorPrimary));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        sessionConfig = new SessionConfig(Splash.this);
```
Comment (mohitathwani) [inline app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/activities/Splash.java]: Can we use ObjectFactory to create SessionConfig? The problem with using "new" operator here is that now this class is tightly coupled with the SessionConfig and this class will not be unit testable in isolation.

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/activities/SheetsList.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_sheets_list);
+
+        // FETCHING ID AND SETTING UP RECYCLER VIEW
+        sheetsListRecyclerView_LinkSheets = findViewById(R.id.sheetsListRecyclerView_LinkSheets);
+        sheetsListRecyclerView_LinkSheets.setLayoutManager(new LinearLayoutManager(SheetsList.this, RecyclerView.VERTICAL, false));
+
+        // INITIALIZE USER MANAGER AND SESSION CONFIG
+        userManager = objectFactory.getUserManager();
+        sessionConfig = new SessionConfig(SheetsList.this);
```
Comment (mohitathwani) [inline app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/activities/SheetsList.java]: Refer to my other comment about SessionConfig

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/adapters/DashboardCardsAdapter.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        holder.availableCollapsedText.setText(getCalculatedValue(model.getAvailableAmount()));
+        holder.spentCollapsedText.setText(getCalculatedValue(model.getSpentAmount()));
+
+        holder.categoriesAvailableCollapsed.setText(String.format("View %d categories", model.getCategoryName().size()));
+    }
+
+    private String getCalculatedValue(List<String> amountList) {
+        double value = 0.00;
+        for (int i = 0; i < amountList.size(); i++) {
+            value += Float.parseFloat(amountList.get(i).replace("$", ""));
```
Comment (mohitathwani) [inline app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/adapters/DashboardCardsAdapter.java]: There are users from around the world using the app. You cannot assume the currency will be always be $. You have to check the users locale and set the currency automatically. Message me on Slack to discuss this issue.

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/fragments/Dashboard.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            @Override
+            protected void onPostExecute(Void aVoid) {
+                super.onPostExecute(aVoid);
+
+                parseFetchedData();
+            }
+        }.execute();
+    }
+
+    private void parseFetchedData() {
```
Comment (mohitathwani) [inline app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/fragments/Dashboard.java]: A View should not be responsible for parsing data. A view should only be responsible for displaying the data. Fetching data, parsing data, etc should be handled by Facility classes.

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/utils/SessionConfig.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    private static final String SharedPref_name = "com.aspirebudgetingmobile.aspirebudgeting_name";
+    private static final String SharedPref_profile = "com.aspirebudgetingmobile.aspirebudgeting_profile";
+    private static final String SharedPref_sheetId = "com.aspirebudgetingmobile.aspirebudgeting_sheetId";
+
+    public SessionConfig(Context context) {
+        this.context = context;
+        sharedPreferences = context.getSharedPreferences(initSharedPref, Context.MODE_PRIVATE);
+    }
+
+    public void setloginStatus(boolean status) {
```
Comment (mohitathwani) [inline app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/utils/SessionConfig.java]: rename function to setLoginStatus

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/utils/SessionConfig.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putBoolean(SharedPref_loginStatus, status);
+        editor.apply();
+    }
+
+    public boolean getloginStatus() {
```
Comment (mohitathwani) [inline app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/utils/SessionConfig.java]: Rename function to getLoginStatus

File: README.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
 # aspirebudgeting_android
-[![Build Status](https://travis-ci.com/aspirebudgetingmobile/aspirebudgeting_android.svg?branch=master)](https://travis-ci.com/aspirebudgetingmobile/aspirebudgeting_android)
+[![Build Status](https://travis-ci.org/akashsarkar188/aspirebudgeting_android.svg?branch=master)](https://travis-ci.com/aspirebudgetingmobile/aspirebudgeting_android)
```
[the remark this answers is not in the document]
Comment (akashsarkar188) [reply README.md]: OKay I will look into this.

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/activities/Splash.java (hunk 1 above)
Comment (akashsarkar188) [reply app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/activities/Splash.java]: Yeah sure, I will remove the param from constructor !

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/adapters/DashboardCardsAdapter.java (hunk 3 above)
Comment (akashsarkar188) [reply app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/adapters/DashboardCardsAdapter.java]: Okay

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/fragments/Dashboard.java (hunk 4 above)
Comment (akashsarkar188) [reply app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/fragments/Dashboard.java]: Sure, I will shift it to an util class

File: app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/utils/SessionConfig.java (hunk 5 above)
Comment (akashsarkar188) [reply app/src/main/java/com/aspirebudgetingmobile/aspirebudgeting/utils/SessionConfig.java]: okay

**Pull request closed without merging**